src/main/java/de/uni_tuebingen/ub/nppm/model/SelektionArealTyp.java

```diff
@@ -7,21 +7,17 @@
 @Table(name = "selektion_arealtyp")
 public class SelektionArealTyp extends SelektionBezeichnung {
     @ManyToMany(mappedBy = "arealTyp")
-    private Set<Einzelbeleg> einzelbeleg = new HashSet<>();
+    private Set<Einzelbeleg> einzelbelege = new HashSet<>();
 
     public void addEinzelbeleg(Einzelbeleg beleg) {
-        this.getEinzelbeleg().add(beleg);
+        this.getEinzelbelege().add(beleg);
     }
 
     public void removeEinzelbeleg(int id) {
-        this.getEinzelbeleg().removeIf(e -> e.getId() == id);
+        this.getEinzelbelege().removeIf(e -> e.getId() == id);
```

Comment:
Einzelbeleg.getId() returns an Integer, so comparing with == will always be false. You should use .equals() instead

Reply:
i have changed the comparison to equals()